test/api_checkpoint_test.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"github.com/alibaba/pouch/test/environment"
+	"github.com/alibaba/pouch/test/request"
+
+	"github.com/go-check/check"
+)
+
+// APIContainerCheckpointSuite is the test suite for container create API.
+type APIContainerCheckpointSuite struct{}
+
+func init() {
+	check.Suite(&APIContainerCheckpointSuite{})
+}
+
+// SetUpTest does common setup in the beginning of each test.
+func (suite *APIContainerCheckpointSuite) SetUpTest(c *check.C) {
+	SkipIfFalse(c, environment.IsLinux)
+	PullImage(c, busyboxImage)
+}
+
+// TestCheckpointCreateAPINonExist tests create a non-existing container checkpoint return error.
+func (suite *APIContainerCheckpointSuite) TestCheckpointCreateAPINonExist(c *check.C) {
+	cname := "TestCheckpointCreateAPINonExist"
+
+	body := request.WithJSONBody(map[string]interface{}{})
+	resp, err := request.Post("/containers/"+cname+"/checkpoints", body)
+	c.Assert(err, check.IsNil)
+
+	CheckRespStatus(c, resp, 404)
+}
+
+// TestCheckpointCreateAPI tests checkpoint create api
+func (suite *APIContainerCheckpointSuite) TestCheckpointCreateAPI(c *check.C) {
```

Comment:
Could we add one test case `TestCheckpointCreateAPIWithExitFlag` to checkpoint and stop one container, then check the stopped status and the checkpoint of this stopped container?